Makefile

```diff
@@ -1,133 +1,155 @@
-	rm build/*.make.yml
-	docker-compose exec -T php composer config repositories.drupal composer https://packages.drupal.org/8
-	docker-compose exec -T php time composer require -o --update-no-dev --no-suggest $(COMPOSER_REQUIRE)
-	make -s front
-	make -s si
+	cp src/composer/composer.json build/;
+ifeq ($(shell test -e src/composer/composer.lock && echo -n yes),yes)
+	cp src/composer/composer.lock build/
+endif
+	make -s composermerge;
+	$(call execute,php,composer install --prefer-dist --optimize-autoloader);
+ifneq ($(shell test -e build/sites/default/settings.php && echo -n yes),yes)
+	$(call execute,php,cp sites/default/default.settings.php sites/default/settings.php);
+	$(call execute,php,sed -i "s/.*file_public_path.*/\$$settings['file_public_path'] =  'docroot\/files';/" sites/default/settings.php);
+endif
 
-si:
-ifeq ($(PROJECT_INSTALL), config)
-	-docker-compose exec -T php time drush si config_installer --db-url=mysql://d8:d8@mysql/d8 --account-pass=admin -y config_installer_sync_configure_form.sync_directory=sync
+ifeq ($(shell test -e site_settings/settings.$(ENV).php && echo -n yes),yes)
+	@echo "We have settings for $(ENV) environment, including it as settings.local.php";
+	$(call execute,php,sed -i "/settings.local.php';/s/# //g" sites/default/settings.php);
+	cp site_settings/settings.$(ENV).php build/settings.local.php; \
+	$(call execute,php,mv settings.local.php sites/default/settings.local.php);
+endif
+	make -s si;
+	$(call execute,php,sh -c "cd docroot && ln -s ../robots.txt && ln -s ../.htaccess");
+
+# Merge several composer.json files.
+composermerge:
+ifeq ($(shell test -e src/composer/composer.$(ENV).json && echo -n yes),yes)
+	@echo "We have local json for $(ENV) environment, including it";
+	cp src/composer/composer.$(ENV).json build/composer.local.json
+	$(call execute,php,composer config extra.merge-plugin.include composer.local.json);
+	$(call execute,php,composer config extra.merge-plugin.recurse true);
+	$(call execute,php,composer config extra.merge-plugin.replace false);
+	$(call execute,php,composer config extra.merge-plugin.merge-dev true);
+	$(call execute,php,composer config extra.merge-plugin.merge-extra true);
+	$(call execute,php,composer config extra.merge-plugin.merge-extra-deep true);
+	$(call execute,php,composer config extra.merge-plugin.merge-scripts true);
 else
-	docker-compose exec -T php drush si $(PROFILE_NAME) --db-url=mysql://d8:d8@mysql/d8 --account-pass=admin -y --site-name="$(SITE_NAME)"
-	docker-compose exec php ash -c "drush locale-check && drush locale-update"
+	$(info "Nothing to merge")
 endif
-	make -s chown
+
+
+update:
+	make -s composermerge; \
+	$(call execute,php,composer update --prefer-dist --optimize-autoloader);
+	$(call execute,php,drush updb -y);
+	$(call execute,php,drush pmu $(MODULES) -y);
+	$(call execute,php,drush en $(MODULES) -y);
+	make -s front; \
+	make -s csim; \
+	make -s trans; \
+	make -s postinstall; \
+	make -s rsync; \
```

Comment:
i think `rsync` must be enabled inside `front` goal, it will be easier to run one command instead of 2 during theming